src/EnergyPlus/ScheduleManager.cc

```diff
@@ -506,12 +517,151 @@ namespace ScheduleManager {
 		//because need a week for each day
 		AddDaySch += NumExternalInterfaceFunctionalMockupUnitExportSchedules; // one day schedule for ExternalInterface
 		// to update during run time
+		
+		CurrentModuleObject = "Schedule:File:Shading";
+		NumCommaFileShading = GetNumObjectsFound( CurrentModuleObject );
+		NumAlphas = 0;
+		NumNumbers = 0;
+		if ( NumCommaFileShading > 1 ) {
+			ShowWarningError( CurrentModuleObject + ": More than 1 occurence of this object found, only first will be used." );
+		}
+
+		NumCSVAllColumnsSchedules = 0;
+
+		if ( NumCommaFileShading != 0 ) {
+			GetObjectItem( CurrentModuleObject, 1, Alphas, NumAlphas, Numbers, NumNumbers, Status, lNumericBlanks, lAlphaBlanks, cAlphaFields, cNumericFields );
+			std::string ShadingSunlitFracFileName = Alphas( 1 );
+			CheckForActualFileName( ShadingSunlitFracFileName, FileExists, TempFullFileName );
+			if ( ! FileExists ) {
+				ShowSevereError( RoutineName + ":\"" + ShadingSunlitFracFileName + "\" not found when External Shading Calculation Method = ImportedShading." );
+				ShowContinueError( "Certain run environments require a full path to be included with the file name in the input field." );
+				ShowContinueError( "Try again with putting full path and file name in the field." );
+				ErrorsFound = true;
+			} else {
+				SchdFile = GetNewUnitNumber();
+				{ IOFlags flags; flags.ACTION( "read" ); gio::open( SchdFile, ShadingSunlitFracFileName, flags ); read_stat = flags.ios(); }
```

Comment:
@XuanLuoLBL This needs to use `TempFullFileName` here because the file could be in a number of different places. I'll fix this.